### ISSUE TITLE
Components from private registry not included in upgradable list

### ISSUE THREAD

**Issue by tdukai** [raineorshine/npm-check-updates#827]
- [X] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [X] I am using the latest version of `npm-check-updates`
- [X] I am using `node >= 10.17`

---------------------------

## Steps to Reproduce
We have a private registry, where our components are stored. The components are replicated to NPM but also marked private.
Node: 14.15.4 LTS, 
NPM: 7.5.4

## Current Behavior
Currently the 11.1.4 and 11.1.3 version not showing our components in the upgradable list:

 @hapi/boom               ^9.1.0  →   ^9.1.1     
 @hapi/hapi              ^20.0.1  →  ^20.1.0     
 @hapi/hoek               ^9.1.
 ajv                     ^6.12.6  →   ^7.1.1     
 fs-extra                 ^9.0.1  →   ^9.1.0     
 hapi-swagger              =13.x  →    =14.x     
 humanize-duration       ^3.24.0  →  ^3.25.1     
 interactjs              ^1.10.0  →  ^1.10.3     
 moment-timezone         ^0.5.31  →  ^0.5.33     
 mongodb                  ^3.6.2  →   ^3.6.4     
 node-schedule            ^1.3.2  →   ^2.0.0     
 sharp                   ^0.26.2  →  ^0.27.1     
 sortablejs              ^1.12.0  →  ^1.13.0     
 upath                    ^2.0.0  →   ^2.0.1     
 uuid                     ^8.3.1  →   ^8.3.2     
 vanilla-picker          ^2.10.1  →  ^2.11.2     
 chai                     ^4.2.0  →   ^4.3.0     
 chokidar                 ^3.4.3  →   ^3.5.1     
 gulp-load-plugins        ^2.0.5  →   ^2.0.6     
 gulp-terser              ^1.4.1  →   ^2.0.1 

## Expected Behavior
The 11.1.1 version properly displaying our private components:

 @hapi/boom               ^9.1.0  →   ^9.1.1     
 @hapi/hapi              ^20.0.1  →  ^20.1.0     
 @hapi/hoek               ^9.1.0  →   ^9.1.1     
 **@unicomp/cloud-storage   ^4.2.3  →   ^4.2.4**     
 **@unicomp/fsdb            ^6.0.1  →   ^6.0.2**     
 **@unicomp/invite          ^1.0.2  →   ^1.1.0**     
 **@unicomp/tools          ^23.0.0  →  ^23.1.2**     
 ajv                     ^6.12.6  →   ^7.1.1     
 fs-extra                 ^9.0.1  →   ^9.1.0     
 hapi-swagger              =13.x  →    =14.x     
 humanize-duration       ^3.24.0  →  ^3.25.1     
 interactjs              ^1.10.0  →  ^1.10.3     
 moment-timezone         ^0.5.31  →  ^0.5.33     
 mongodb                  ^3.6.2  →   ^3.6.4     
 node-schedule            ^1.3.2  →   ^2.0.0     
 sharp                   ^0.26.2  →  ^0.27.1     
 sortablejs              ^1.12.0  →  ^1.13.0     
 upath                    ^2.0.0  →   ^2.0.1     
 uuid                     ^8.3.1  →   ^8.3.2     
 vanilla-picker          ^2.10.1  →  ^2.11.2     
 chai                     ^4.2.0  →   ^4.3.0     
 chokidar                 ^3.4.3  →   ^3.5.1     
 gulp-load-plugins        ^2.0.5  →   ^2.0.6     
 gulp-terser              ^1.4.1  →   ^2.0.1

**Comment by raineorshine** [comment]
Does `npx pacote packument PACKAGE_NAME` work? That will help confirm if the problem is in `ncu` or elsewhere.

> Currently the 11.1.4 and 11.1.3 version not showing our components in the upgradable list:

> The 11.1.1 version properly displaying our private components:

I'm a bit confused, as I don't see `11.1.1`, `11.1.3`, nor `11.1.4` in your dependencies. What version is currently specified, and what is the version published to the `latest` tag?

**Comment by cbogwill** [comment]
@raineorshine I believe they mean version 11.1.1, 11.1.3 and 11.1.4 of ncu.  I'm seeing the same issues when upgrading to 11.1.4 of ncu where I have a local package that was on 3.0.47, and I know there was 3.0.58 available.  running ncu showed no update available.

**Comment by cbogwill** [comment]
@raineorshine not sure this helps, but here is output of both ncu and npm outdated, as well as npx pacote packument

```
C:\projects\email-listener>ncu
Checking C:\projects\email-listener\package.json
[====================] 21/21 100%

 form-data  ^3.0.0  →  ^4.0.0

Run ncu -u to upgrade package.json

C:\projects\email-listener>npm outdated
Package                Current  Wanted  Latest  Location                            Depended by
form-data                3.0.0   3.0.1   4.0.0  node_modules/form-data              email-listener
riley                   2.2.73  2.2.73   3.0.3  node_modules/riley                  email-listener
riley-openapi-request   3.0.47  3.0.58  3.0.58  node_modules/riley-openapi-request  email-listener
riley-runas-oidc        1.0.41  1.0.41   3.0.0  node_modules/riley-runas-oidc       email-listener
```

```
C:\projects\email-listener>npx pacote packument riley-openapi-request
HttpErrorGeneral: 404 Not Found - GET https://registry.npmjs.org/riley-openapi-request - Not found
    at C:\Users\cxb4553\AppData\Local\npm-cache\_npx\d68c951c7177ed18\node_modules\npm-registry-fetch\check-response.js:123:15
    at processTicksAndRejections (internal/process/task_queues.js:93:5) {
  headers: [Object: null prototype] {
    date: [ 'Thu, 18 Feb 2021 16:13:17 GMT' ],
    'content-type': [ 'application/json' ],
    'content-length': [ '21' ],
    connection: [ 'keep-alive' ],
    'set-cookie': [
      '__cfduid=db42030296caa8763b96915db4b0589ea1613664797; expires=Sat, 20-Mar-21 16:13:17 GMT; path=/; domain=.npmjs.org; HttpOnly; SameSite=Lax'
    ],
    'cf-ray': [ '623908db2f0e26d3-MSP' ],
    age: [ '69' ],
    vary: [ 'Accept-Encoding' ],
    'cf-cache-status': [ 'HIT' ],
    'cf-request-id': [ '085783dcf9000026d3e9161000000001' ],
    'expect-ct': [
      'max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"'
    ],
    server: [ 'cloudflare' ],
    'x-fetch-attempts': [ '1' ]
  },
  statusCode: 404,
  code: 'E404',
  method: 'GET',
  uri: 'https://registry.npmjs.org/riley-openapi-request',
  body: { error: 'Not found' },
  pkgid: 'riley-openapi-request@latest'
}
npm ERR! code 1
npm ERR! path C:\projects\email-listener
npm ERR! command failed
npm ERR! command C:\windows\system32\cmd.exe /d /s /c pacote packument riley-openapi-request

npm ERR! A complete log of this run can be found in:
```

**Comment by mima0815** [comment]
I checked this with a project with private repositories here on my side:

performing the command `npx pacote packument @myscope/mypackage` locally always returns this:

```
npx: installed 158 in 28.856s
HttpErrorGeneral: 404 Not Found - GET https://registry.npmjs.org/@myscope%2fmypackage - Not found
    at /Users/user/.npm/_npx/52048/lib/node_modules/pacote/node_modules/npm-registry-fetch/check-response.js:123:15
    at processTicksAndRejections (internal/process/task_queues.js:93:5) {
  headers: [Object: null prototype] {
    date: [ 'Thu, 18 Feb 2021 16:02:30 GMT' ],
    'content-type': [ 'application/json' ],
    'content-length': [ '21' ],
    connection: [ 'keep-alive' ],
    'set-cookie': [
      '__cfduid=d0ecf4e259a20aa5bac29cbd76511f6f51613664150; expires=Sat, 20-Mar-21 16:02:30 GMT; path=/; domain=.npmjs.org; HttpOnly; SameSite=Lax'
    ],
    'cf-request-id': [ '085779fa880000e674cf2d5000000001' ],
    'expect-ct': [
      'max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"'
    ],
    vary: [ 'Accept-Encoding' ],
    server: [ 'cloudflare' ],
    'cf-ray': [ '6238f90a7e62e674-LHR' ],
    'x-fetch-attempts': [ '1' ]
  },
  statusCode: 404,
  code: 'E404',
  method: 'GET',
  uri: 'https://registry.npmjs.org/@myscope%2fmypackage ',
  body: { error: 'Not found' },
  pkgid: '@myscope/mypackage@latest'
}
```

it is possible to get `ncu` working, i adding to the normal registry the scoped registry too.

Without the scoped one it was working until version 11.1.2 or 11.1.1

```
registry = http://myregistry/npm
@myscope:registry = http://myregistry/npm
```

**Comment by raineorshine** [comment]
> @raineorshine I believe they mean version 11.1.1, 11.1.3 and 11.1.4 of ncu. I'm seeing the same issues when upgrading to 11.1.4 of ncu where I have a local package that was on 3.0.47, and I know there was 3.0.58 available. running ncu showed no update available.

Ah, that makes sense. It must be a regression at some point then.

There are 31 commits between https://github.com/raineorshine/npm-check-updates/compare/v11.1.1...v11.1.3

If someone can create a repo that reproduces or adds me to a repo that reproduces I can troubleshoot further. Feel free to clone `npm-check-updates` and investigate yourself as well. Once in a reproducible state, I would try to identity the exact commit of the regression.

**Comment by tdukai** [comment]
> Does `npx pacote packument PACKAGE_NAME` work? That will help confirm if the problem is in `ncu` or elsewhere.
> 
> > Currently the 11.1.4 and 11.1.3 version not showing our components in the upgradable list:
> 
> > The 11.1.1 version properly displaying our private components:
> 
> I'm a bit confused, as I don't see `11.1.1`, `11.1.3`, nor `11.1.4` in your dependencies. What version is currently specified, and what is the version published to the `latest` tag?

Sorry I meant that NCU versions 11.1.1 works properly, later versions do not.

**Comment by raineorshine** [comment]
I think I fixed it. Will someone try out `v11.1.9`?

**Comment by cbogwill** [comment]
@raineorshine I have tried it and can confirm it works.  Thanks!

**Comment by muzuiget** [comment]
Come from [#837](https://github.com/raineorshine/npm-check-updates/issues/831#issuecomment-784664789), I can confirm the ~/.npmrc `registery` issue fixed in `v11.1.9`, thanks.

**Comment by tdukai** [comment]
Confirmed, 11.1.9 works as required. Thank you very much for the quick fix! Awesome job!
